flang/CMakeLists.txt

```
@@ -436,6 +436,10 @@ if (LLVM_COMPILER_IS_GCC_COMPATIBLE)
    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -fno-semantic-interposition")
   endif()
 
+  # gcc requires this flag in order for precompiled headers to work with ccache
+  if (NOT CMAKE_CXX_COMPILER_ID MATCHES Clang AND NOT CMAKE_DISABLE_PRECOMPILE_HEADERS)
```

Since this flag is required by GCC, can we test that the compiler is GCC?

```suggestion
  # GCC requires this flag in order for precompiled headers to work with ccache
  if (CMAKE_CXX_COMPILER_ID MATCHES GCC AND NOT CMAKE_DISABLE_PRECOMPILE_HEADERS)
```

---

I was following what the rest of the file was doing in matching on "not clang" in this context, but yes of course. Done.